hub/cli/auth.py

```diff
@@ -15,24 +15,26 @@
 
 
 @click.command()
-@click.option("--username", "-u", default=None, help="Your Activeloop username")
-@click.option("--password", "-p", default=None, help="Your Activeloop password")
+@click.option("--username", "-u", default=None, help="Your Activeloop Username")
+@click.option("--password", "-p", default=None, help="Your Activeloop Password")
 def login(username: str, password: str):
     """Log in to Activeloop"""
-    click.echo("Login to Activeloop Hub using your credentials.")
-    click.echo(
-        "If you don't have an account, register by using 'activeloop register' command or by going to "
-        f"{HUB_REST_ENDPOINT}/register."
-    )
-    username = username or click.prompt("Username")
+    # If any/all argument values are None, request the user to enter credentials
+    if all([username, password]) == False:
+        click.echo("Login to Activeloop Hub using your credentials.")
+        click.echo(
+            "If you don't have an account, register by using the 'activeloop register' command or by going to "
+            f"{HUB_REST_ENDPOINT}/register."
+        )
+        username = click.prompt("Username")
```

Comment:
Maybe don't prompt if already given in command line? `username = username or click.prompt("Username")`

Reply:
That was present in the previous implementation. I feel it's necessary to separate the two expressions because we are checking here if the user has entered command line arguments or not. If user has provided command line arguments, then there wouldn't be the message "Login to Activeloop Hub using your credentials", it would directly just let the user know if your credentials are right or not. If the command line arguments aren't provided, then it would display that message and ask the user to manually enter the credentials. The one line statement you provided does not perform that check explicitly; the message above gets printed even if you enter command line arguments.

Reply:
Hmm.. maybe a bit nit picky but what about:

```python
def login(username: str, password: str):
    if not username:
        # No creds in command line args. (Ignore password even if provided in this case)
        password = None  # (Triggers password prompt after username prompt)
        click.echo("Login to Activeloop Hub using your credentials.")
        click.echo("If you don't have an account,........")
        username = click.prompt("Username")
    if not password:
        password = click.prompt("Enter password for %s" & username)
    ##############
```

This handles the case where provides just the username without password a bit more gracefully.

Reply:
Sounds great, but one question: wouldn't we have to explicitly code the cases where another command line argument is provided, but the rest are not? For example, here, if the user just decides to provide the password and not the username (very highly unlikely, but still it's a case). And in the case of `activeloop register`, where there are three command line arguments, we'd have to make explicit cases for each of them right? If the user has entered the username/password but not email, and so on. Wouldn't this approach be better? If not all necessary command line arguments have been passed, redirect the user to manually enter the credentials, so there can be lesser chances of errors as well.